modules/ding_nodelist/ding_nodelist.admin.inc

```diff
@@ -0,0 +1,186 @@
+  }
+
+  $form['ding_nodelist_node_types'] = array(
+    '#type' => 'checkboxes',
+    '#title' => t('Available content types'),
+    '#options' => $options,
+    '#default_value' => variable_get('ding_nodelist_node_types', array()),
+    '#description' => t('Choose content types avaialable for display'),
+  );
+
+  // Options as defined by panels/plugins/cache/simple.inc
+  $options = drupal_map_assoc(
+    array(
+      15, 30, 60, 120, 180, 240, 300, 600, 900, 1200, 1800, 3600,
+      7200, 14400, 28800, 43200, 86400, 172800, 259200, 345600, 604800),
+    'format_interval');
+  $form['ding_nodelist_cache_default_time'] = array(
+    '#type' => 'select',
+    '#options' => $options,
+    '#title' => t('Default cache life time'),
+    '#default_value' => variable_get('ding_nodelist_cache_default_time', NL_CACHE_LIFETIME),
+    '#description' => t('Lists will be cached by default for this period of time (seconds)'),
+  );
+
+  $form['ding_nodelist_autoscroll_delay'] = array(
+    '#type' => 'fieldset',
+    '#collapsible' => FALSE,
+    '#title' => t('Autoscroll delay'),
+    '#description' => t('The delay to show an item for various nodelist widgets (in milliseconds).'),
+  );
+
+  $autoscrolling_widgets = array(
+    NL_WIDGET_CAROUSEL => array('Carousel', 3000),
+    NL_WIDGET_HOR_ACCORDION => array('Horizontal accordion', 3000),
+  );
+
+  foreach ($autoscrolling_widgets as $widget => $conf) {
+    $title = isset($conf[0]) ? check_plain($conf[0]) : '';
+    $form['ding_nodelist_autoscroll_delay']['ding_nodelist_' . $widget . '_delay'] = array(
+      '#type' => 'textfield',
+      '#title' => filter_xss($title),
+      '#default_value' => variable_get('ding_nodelist_' . $widget . '_delay', $conf[1]),
+      '#size' => 5,
+    );
+  }
+
+  return system_settings_form($form);
+}
+
+/**
+ * Template settings form.
+ */
+function ding_nodelist_templates_form($form, &$form_state) {
+
+  $form['templates'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Templates'),
+    '#tree' => TRUE,
+    '#theme' => '_ding_nodelist_templates_table',
+  );
+
+  $templates = _ding_nodelist_get_templates();
+
+  foreach ($templates as $template_name => $template) {
+    $form['templates'][$template_name]['template_name'] = array(
+      '#type' => 'hidden',
+      '#value' => $template_name,
+    );
+
+    $form['templates'][$template_name]['title'] = array(
+      '#type' => 'textfield',
+      '#default_value' => isset($template['title']) ? $template['title'] : '',
+      '#size' => 40,
+    );
+
+    $ct = node_type_get_names();
+    $form['templates'][$template_name]['content_type'] = array(
+      '#markup' => isset($ct[$template['content_type']]) ?
+      $ct[$template['content_type']] : t('Missing (@ctname)', array('@ctname' => $template['content_type'])),
+    );
+
+    $widgets = _ding_nodelist_get_widget_types();
+    $form['templates'][$template_name]['widget'] = array(
+      '#type' => 'select',
+      '#empty_option' => t('- select -'),
```

Comment:
Why do we set this here? [Drupal provides a default](https://api.drupal.org/api/drupal/developer%21topics%21forms_api_reference.html/7.x#empty_option).